util/reggen/uvm_reg.sv.tpl

```diff
@@ -138,20 +138,24 @@ package ${block.name}_ral_pkg;
       super.new(name, has_coverage);
     endfunction : new
 
-    virtual function void build(uvm_reg_addr_t base_addr);
+    virtual function void build(uvm_reg_addr_t base_addr, csr_utils_pkg::csr_excl_item csr_excl);
       // create default map
       this.default_map = create_map(.name("default_map"),
                                     .base_addr(base_addr),
                                     .n_bytes(${block.width//8}),
                                     .endian(UVM_LITTLE_ENDIAN));
+      if (csr_excl == null) begin
```

Comment:
Thank you for the detailed suggestion, modified as you recommended.

Reply:
Thank you for the suggestion, and I agree the implementation was a little bit weird. But seems like your proposal has some issue as well:
1). local is not supported : 'local::' is not allowed in context: 'not in randomize-with constraint'
2). Later code will add exclusion to `csr_excl`, if here we only create the global `this.csr_excl`, this will cause compile error due to null pointer object.

How about change the naming to make is more clear?
```
  if (local_csr_excl == null) begin
    local_csr_excl = csr_utils_pkg::...;
    this.csr_excl = local_csr_excl;
  end
```

Reply:
I modified my original comment - i think that should work.

Reply:
Aah i see....nevermind. Stick to existing code.